scripts/post-install.sh

```diff
@@ -45,29 +44,42 @@ fi
 # Distribution-specific logic
 if [[ -f /etc/redhat-release ]]; then
     # RHEL-variant logic
-    which systemctl &>/dev/null
-    if [[ $? -eq 0 ]]; then
-	install_systemd
+    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+        install_systemd
     else
-	# Assuming sysv
-	install_init
-	install_chkconfig
+        # Assuming SysVinit
+        install_init
+        # Run update-rc.d or fallback to chkconfig if not available
+        if which update-rc.d >/dev/null; then
+            install_update_rcd
+        else
+            install_chkconfig
+        fi
     fi
 elif [[ -f /etc/debian_version ]]; then
     # Debian/Ubuntu logic
-    which systemctl &>/dev/null
-    if [[ $? -eq 0 ]]; then
-	install_systemd
+    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+        install_systemd
     else
-	# Assuming sysv
-	install_init
-	install_update_rcd
+        # Assuming SysVinit
+        install_init
+        # Run update-rc.d or fallback to chkconfig if not available
+        if which update-rc.d >/dev/null; then
```

Comment:
All calls to `which` should be updated with `&>/dev/null`, not `>/dev/null`. On CentOS, I get this on a vanilla installation:

```
$ sudo rpm -i influxdb-1.2.1~rc3~cab47cd.x86_64.rpm
which: no update-rc.d in (/sbin:/bin:/usr/sbin:/usr/bin:/usr/X11R6/bin)
which: no update-rc.d in (/sbin:/bin:/usr/sbin:/usr/bin:/usr/X11R6/bin)
```